fix: host not generating its own local client disconnect notification

NoelStephensUnity opened the pull request:
This fix assures a host will invoke `OnClientDisconnectCallback` for its local client when it is internally shutting down.

fix: #2789

## Changelog

- Fixed: Issue where the host was not invoking `OnClientDisconnectCallback` for its own local client when internally shutting down.

## Testing and Documentation

- Includes integration test update.
- No documentation changes or additions were necessary.

<!--  Uncomment and mark items off with a * if this PR deprecates any API:
### Deprecated API
- [ ] An `[Obsolete]` attribute was added along with a `(RemovedAfter yyyy-mm-dd)` entry.
- [ ] An [api updater] was added.
- [ ] Deprecation of the API is explained in the CHANGELOG.
- [ ] The users can understand why this API was removed and what they should use instead.
-->

NoelStephensUnity committed:
fix

Assure the host generates a notification for the local host client no longer being connected when it shuts itself down.

NoelStephensUnity committed:
test

Include modification to the DisconnectTest to validate this fix.

NoelStephensUnity committed:
update

updating the change log

NoelStephensUnity committed:
update

fixing improper name for callback

NoelStephensUnity committed:
style

NoelStephensUnity committed:
test

Rewrote the SenderIdTest to be a bit more flexible with regard to waiting for the client disconnect event and not failing if it received anything afterwards.

NoelStephensUnity committed:
style

whitespace removal

ShadauxCat commented on testproject/Assets/Tests/Runtime/SenderIdTests.cs:
```diff
@@ -89,36 +90,34 @@ public IEnumerator WhenSendingMessageFromClientToServer_SenderIdIsCorrect()
-            }
-            m_ServerNetworkManager.OnClientDisconnectCallback += firstCallback;
+            m_ClientsDisconnected.Clear();
+            m_ServerNetworkManager.OnClientDisconnectCallback += OnClientDisconnected;
```
Can you also add a check for this behavior in `OnConnectionEvent`? `InvokeOnClientDisconnectCallback` should invoke both of them.

NoelStephensUnity replied:
Also discovered an issue with StopOneClient while adding the test... so it was a 2 4 1.
Validated the ConnectionEvent.ClientDisconnected triggers under both scenarios and fixed the integration test method.
👍

NoelStephensUnity committed:
test - fix

Adding check for ConnectionEvent.ClientDisconnected for both test types.
Fixing issue with StopOneClient destroying the NetworkManager when destroy is set to false.

netcode-ci-service committed:
Merge develop into fix/host-on-client-disconnect-local-client

netcode-ci-service committed:
Merge develop into fix/host-on-client-disconnect-local-client

netcode-ci-service committed:
Merge develop into fix/host-on-client-disconnect-local-client
